chia/consensus/blockchain.py

```diff
@@ -271,9 +271,14 @@ async def add_block(
             block,
             None,
         )
-        # Always add the block to the database
-        async with self.block_store.db_wrapper.writer():
-            try:
+
+        # in case we fail and need to restore the blockchain state, remember the
+        # peak height
+        previous_peak_height = self._peak_height
+
+        try:
+            # Always add the block to the database
+            async with self.block_store.db_wrapper.writer():
```

Comment:
the `try` was swapped with `async with`, that's why this looks a bit messy. Viewing this ignoring whitespace makes it a bit easier to see what happened.